src/megaapi_impl.cpp

```diff
@@ -18186,16 +18186,37 @@ void MegaApiImpl::sendPendingRequests()
 {
     int nextTag = 0;
 
-    while(true)
+    SdkMutexGuard g(sdkMutex);
+    
+    // For multiple consecutive requests of the same type (eg. remove transfer) this committer will put all the database activity into a single commit
+    DBTableTransactionCommitter committer(client->tctable);
+    int lastRequestType = -1;
+    int lastRequestConsecutive = 0;
+
+
+    while(MegaRequestPrivate *request = requestQueue.front())
     {
-        sdkMutex.lock();
 
-        MegaRequestPrivate *request = requestQueue.pop();
-        if (!request)
+        // also we avoid yielding for consecutive transaction cancel operations (we used to yeild every time, but we need to keep the sdkMutex lock while the database transaction is ongoing)
+        if ((lastRequestType == -1 || lastRequestType == request->getType()) && lastRequestConsecutive < 1024)
         {
+            ++lastRequestConsecutive;
+        }
+        else
+        {
+            committer.commitNow();
             sdkMutex.unlock();
+            yield();
+            sdkMutex.lock();
+            lastRequestConsecutive = 0;
+        }
+
+        request = requestQueue.pop();
+        if (!request)
```

Comment:
If the `front()` didn't return `null`, the `pop()` will neither. This check is unnecessary.

Reply:
why not ` while(MegaRequestPrivate *= requestQueue.pop())` ?

Reply:
Yes, I think that arrangement was left over from a version where we might exit the loop before all requests were processed (to give the SDK a go periodically on this thread, but we decided against that for now in a long-ago review, I think).    I'll make this simplification, thank you both for the review!  cheers